routers/api/v1/repo/file.go

```diff
@@ -45,3 +45,27 @@ func GetArchive(ctx *context.APIContext) {
 
 	repo.Download(ctx.Context)
 }
+
+func GetEditorconfig(ctx *context.APIContext) {
+	if !ctx.Repo.HasAccess() {
+		ctx.Status(404)
+		return
+	}
+	ec, err := ctx.Repo.GetEditorconfig()
+	if err != nil {
+		if git.IsErrNotExist(err) {
+			ctx.Handle(404, "ErrEditorconfigNotExists", err)
+		} else {
+			ctx.Handle(500, "ErrGettingEditorconfig", err)
+		}
+		return
+	}
+
+	fileName := ctx.Params("*")
```

Comment:
I'm passing the default branch in the URL because this is required by the `RepoRef()` in the router.

Initially I tried without `RepoRef()` but had a problem that way. Anyway, in the JavaScript, the filename in concatenated in the URL.

I'll reconsider some decisions the update.